torch/utils/data/dataloader.py

```diff
@@ -141,11 +142,15 @@ def __init__(self, loader):
                 self._put_indices()
 
     def __len__(self):
-        return int(math.ceil(len(self.sampler) / float(self.batch_size)))
+        if self.drop_last:
+            return len(self.sampler) // self.batch_size
+        else:
+            return len(self.sampler) // self.batch_size + 1
```

Comment:
This is not a correct formula for `ceil` (it's invalid in case `len(self.sampler) % self.batch_size == 0`). If you really want to do that this would be correct: `(len(self.sampler) + self.batch_size - 1) // self.batch_size`.